### PR TITLE
Add support for LexicalHandler.startEntity and endEntity

### PR DESCRIPTION
**Pull request by philippn**
This is the pull request for issue #209

It also adds a test case.

One thing that I find problematic at the moment is that it needs an extra configuration on the STAX factory. It would be better if this would either be the default behaviour -- which is a breaking change but might make sense since other SAX parsers behave that way too -- or at least could be controlled via a feature flag.

### CODE REVIEW

**Comment by cowtowncoder** [comment]
Ok, so looks good (with brief look). But one thing I'll need before merging is CLA, from:

https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf

(used for Woodstox too, not just Jackson)

and it's only needed before the first contribution. The usual way is to print it, fill & sign, scan/photo, email to `cla` at fasterxml dot com. Once I get that I can do final review and merge the fix.

**Comment by philippn** [comment]
Thanks for considering it. I have sent you the CLA.

**Comment by cowtowncoder** [comment]
CLA received, can proceed later tonight.

**Review by cowtowncoder — APPROVED** [review]

**Comment by cowtowncoder** [comment]
@philippn Ok, I will merge this now.

As to the part about awkward configurability: I am open to suggestions here.
Timing is bit unfortunate, as 7.0.0 was released quite recently -- change of defaults would have worked quite nicely if it was part of 7.x change. Less so for minor version (would need to go in 7.1.0). And of course default would just apply to SAX side.

Still: I am open to a follow-up PR for doing just that (or, if you want, feature flag to change it directly). I probably won't have time to look into that myself, but I'll always try to help with contributions so others can help.

**Pull request merged**
